### PR TITLE
FI-473: Make TestingInstance#patient_id deterministic

### PR DESCRIPTION
**Pull request by Jammjammjamm**
This PR adds a `created_at` timestamp to `ResourceReference` and makes `TestingInstance#patient_id` retrieve the id of the Patient reference which was created first. Previously, the first Patient reference was chosen from a list sorted by a randomly generated id, so the id returned was non-deterministic when multiple Patient references were present.

**Submitter:**
- [x] This pull request describes why these changes were made
- [x] Internal ticket for this PR: https://oncprojectracking.healthit.gov/support/browse/FI-473
- [x] Internal ticket links to this PR
- [x] Internal ticket is properly labeled (Community/Program)
- [x] Internal ticket has a justification for its Community/Program label
- [x] Code diff has been reviewed for extraneous/missing code
- [x] Tests are included and test edge cases
- [x] Tests/code quality metrics have been run locally and pass

**Reviewer 1:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure
      where appropriate, and accomplishes the task's purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

**Reviewer 2:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure
      where appropriate, and accomplishes the task's purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

### CODE REVIEW

File: test/unit/test_instance_test.rb [hunk 1]
```diff
@@ -74,4 +74,26 @@
       assert !@instance.fhir_version_match?(fhir_versions)
     end
   end
+
+  describe '#patient_id' do
+    it 'returns the id of the Patient reference which was created first' do
```
[the remark this answers is not in the document]
Comment (Jammjammjamm) [reply test/unit/test_instance_test.rb]: It would, but I'm not sure if it's possible and I feel like this is good enough.

File: test/sequence/medication_order_sequence_test.rb [hunk 2]
```diff
@@ -40,23 +40,24 @@ def setup
...
-                                                     client_endpoint_key: Inferno::SecureRandomBase62.generate(32),
-                                                     client_id: SecureRandom.uuid,
-                                                     selected_module: 'argonaut',
-                                                     oauth_authorize_endpoint: 'http://oauth_reg.example.com/authorize',
-                                                     oauth_token_endpoint: 'http://oauth_reg.example.com/token',
-                                                     scopes: 'launch openid patient/*.* profile',
-                                                     token: 99_897_979)
-
-    @instance.save! # this is for convenience.  we could rewrite to ensure nothing gets saved within tests.
+    @instance = Inferno::Models::TestingInstance.create(
```
[the remark this answers is not in the document]
Comment (Jammjammjamm) [reply test/sequence/medication_order_sequence_test.rb]: Yes. They don't get timestamps until they're persisted. I'm preferring to make the creation explicit rather than depending on `TestingInstance#save` persisting all of its associations.
